File: Source/Aladdin.CAPI.ANSI.PKCS11.Java/src/aladdin/capi/ansi/pkcs11/keyx/rsa/oaep/Encipherment.java
package aladdin.capi.ansi.pkcs11.keyx.rsa.oaep;
import aladdin.pkcs11.*;
import aladdin.pkcs11.jni.*;
import aladdin.capi.*;
import aladdin.capi.pkcs11.*;
import java.io.*; 
    
///////////////////////////////////////////////////////////////////////////
// Алгоритм зашифрования RSA OAEP
///////////////////////////////////////////////////////////////////////////
public class Encipherment extends aladdin.capi.pkcs11.Encipherment
{
    // идентификатор алгоритмов и метка
    private final long hashAlg; private final long mgf; private final byte[] sourceData;
    
    // конструктор
	public Encipherment(Applet applet, long hashAlg, long mgf, byte[] sourceData) throws IOException
    { 
        // сохранить переданные параметры
        super(applet); this.hashAlg = hashAlg; this.mgf = mgf; this.sourceData = sourceData; 
    } 
	// параметры алгоритма
    @Override protected Mechanism getParameters(Session session, IParameters parameters)
	{
        // указать параметры алгоритма
        CK_RSA_PKCS_OAEP_PARAMS oaepParameters = new CK_RSA_PKCS_OAEP_PARAMS(hashAlg, mgf, sourceData); 
        
        // вернуть параметры алгоритма
        return new Mechanism(API.CKM_RSA_PKCS_OAEP, oaepParameters); 
	}
}

File: Source/Aladdin.CAPI.ANSI.PKCS11/Keyx/RSA/OAEP/Encipherment.cs
using System;
using Aladdin.PKCS11; 

namespace Aladdin.CAPI.ANSI.PKCS11.Keyx.RSA.OAEP
{
    ///////////////////////////////////////////////////////////////////////////
    // Алгоритм зашифрования RSA OAEP
    ///////////////////////////////////////////////////////////////////////////
    public class Encipherment : CAPI.PKCS11.Encipherment
    {
        // идентификаторы алгоритмов и метка
        private ulong hashAlg; private ulong mgf; private byte[] sourceData; 
    
        // конструктор
	    public Encipherment(CAPI.PKCS11.Applet applet, 
            ulong hashAlg, ulong mgf, byte[] sourceData) : base(applet)
        { 
            // сохранить переданные параметры
            this.hashAlg = hashAlg; this.mgf = mgf; this.sourceData = sourceData; 
        } 
	    // параметры алгоритма
        protected override Mechanism GetParameters(
            CAPI.PKCS11.Session session, IParameters parameters)
	    {
            // указать параметры алгоритма
            Parameters.CK_RSA_PKCS_OAEP_PARAMS oaepParameters = 
                new Parameters.CK_RSA_PKCS_OAEP_PARAMS(hashAlg, mgf, sourceData); 

            // вернуть параметры алгоритма
            return new Mechanism(API.CKM_RSA_PKCS_OAEP, oaepParameters); 
	    }
    }
}
